src/devices/cpu/tms34010/34010ops.hxx

```diff
@@ -2395,7 +2399,40 @@ void tms340x0_device::rpix_b(uint16_t op)
 void tms340x0_device::setcdp(uint16_t op)
 {
 	if (!m_is_34020) { unimpl(op); return; }
-	logerror("020:setcdp\n");
+	off_t dptch = DPTCH();
+
+	// logerror("020:setcdp: PC=%x, DPTCH()=%x\n", m_pc, dptch);
+
+	// Check whether we're dealing with an even number
+	if ((dptch & 1) == 0) {
+		switch(population_count_32(dptch))
+		{
+			// .. only single bit set, pitch is power of two!
+			case 1:
+			{
+				uint32_t val = log2(dptch);
+				m_convdp = ~val & 0x1f;
+				COUNT_CYCLES(4);
+				return;
+			}
+			// .. two bits, we can decompose it to sum of two power of two numbers
+			case 2:
+			{
+				uint8_t val1 = count_leading_zeros(dptch) - 1;
+				uint8_t val2 = count_leading_zeros(dptch & ~(1 << val1)) - 1;
+
+				val1 = ~val1 & 0x1f;
+				val2 = ~val2 & 0x1f;
```

Comment:
Isn't this equivalent to `31 - x`?  It's going to be easier for the compiler to optimise it if you do this as a subtraction integrated into the operation above.